elastic/kibana
# [Dashboard/Reporting] Removal of Reporting injected CSS and JS

## Summary

This PR demonstrates how we could update the reporting integration in such a way that we no longer depend on the following functionality (example links are to the print layout):

https://github.com/elastic/kibana/blob/f3dc980bb92d450b45bf344f4ae10ef11ff864ed/x-pack/plugins/reporting/server/lib/layouts/print_layout.ts#L52

https://github.com/elastic/kibana/blob/f3dc980bb92d450b45bf344f4ae10ef11ff864ed/x-pack/plugins/reporting/server/lib/layouts/print_layout.ts#L32

The proposal here is to migrate all responsibility of layout to Dashboard while supporting legacy and new V2 export types.

## Notable implementation details

* In order for old reports (i.e., those not using locators) to continue working we use the `screenshotMode` service to pass along not just screenshot mode but also what layout we are intending to capture a screenshot for. The thinking is that this will function as relatively cheap tech debt that enables us to remove the larger tech debt of reporting handling layout/styling of page elements.
* This approach still uses the "print" layout. We can consider using the "canvas" approach instead which would mean dashboard is responsible for moving visualisations onto individual pages. This was avoided for the first round of implementation as this version demonstrates how we can ensure ONLY visualisations are present (i.e., not rendering unnecessary elements, see comment https://github.com/elastic/kibana/pull/111958#issuecomment-922914170) when capturing a print screenshot.

## Screenshot

<img width="1401" alt="Screenshot 2021-09-13 at 15 18 24" src="https://user-images.githubusercontent.com/8155004/133100204-860fc9e0-7e1f-48c8-bc45-c05a491ec052.png">

## How to test

1. Start Kibana & ES
2. Import some sample data
3. Go to dashboard and open one of the sample dashboards
4. Generate a report (non-page optimized)
5. Generate a report (page-oprtimized)
6. View both reports
7. (Optional) See https://github.com/jloleysens/kibana/pull/6 for instructions on how to view the "print" layout in Kibana

### Checklist

- [x] Figure out how `captureConfig` can be respected by dashboard in the new `PrintViewport` component. We decided to move the captureConfig config to `ununsed` status as this is not a very useful/usable setting to expose to users.
- [x] Try to create a new page route for "share" captures that the dashboard locator can send reporting to
